Makefile

```diff
@@ -8,9 +8,9 @@ bindir = $(prefix)/bin
 SEQAN_DIR = SeqAn-1.1
 SEQAN_INC = -I $(SEQAN_DIR)
 INC = $(SEQAN_INC) -I third_party
-CPP = g++
-CXX = $(CPP)
-CC = gcc
+CPP ?= g++
+CXX ?= $(CPP)
+CC ?= gcc
```

Comment:
I think this part is useful. Not only when building by `clang` but also building by `gcc-N` compiler.